superset-frontend/src/views/CRUD/data/dataset/AddDataset/Footer/index.tsx

```diff
@@ -104,7 +100,8 @@ function Footer({
         if (typeof response === 'number') {
           logEvent(LOG_ACTIONS_DATASET_CREATION_SUCCESS, datasetObject);
           // When a dataset is created the response we get is its ID number
-          goToPreviousUrl();
+          window.location.href =
```

Comment:
Previously, users could create a dataset as part of the adding a chart flow and still be on that page when the modal was closed, but with this change they'll need to re-navigate there.  This feels like maybe too big of a UX regression to me – could we implement a more flexible system for redirecting on submit/cancel?

Reply:
I feel like when you finish creating a dataset, I think that the figma had you going to the edit flow. Which is obviously not created yet, maybe for the time being it should redirect you back to where you where?

Reply:
Oops sorry, I missed that in the designs – if we're intentionally changing that flow so it always ends you at the Edit Dataset view no matter where the user started then please disregard.  I'm fine with how it is now (redirect to the dataset list) as a temporary solution but @AAfghahi's suggestion works too.

Reply:
I think originally we were rerouting to the dataset list so the user can see the dataset they just created. I'm gonna bring in @yousoph for this one, should the user be redirected to the dataset list or their previous page after creating a dataset?

Reply:
Confirmed from @yousoph: for this step, redirect to Add Chart page on successful dataset create.  May possibly change button text as well.

Reply:
Fixed in [`this commit`](https://github.com/apache/superset/pull/22610/commits/d2903e307228270c40270088b7a83d64ae13e545);